src/Meshes/mesh.ts

```diff
@@ -2398,6 +2398,229 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
+                    deltaPosition.y = (positions[3 * tempIndices[k][k] + 1] - positions[3 * tempIndices[k][0] + 1]) / k;
+                    deltaPosition.z = (positions[3 * tempIndices[k][k] + 2] - positions[3 * tempIndices[k][0] + 2]) / k;
+                    deltaNormal.x = (normals[3 * tempIndices[k][k]] - normals[3 * tempIndices[k][0]]) / k;
+                    deltaNormal.y = (normals[3 * tempIndices[k][k] + 1] - normals[3 * tempIndices[k][0] + 1]) / k;
+                    deltaNormal.z = (normals[3 * tempIndices[k][k] + 2] - normals[3 * tempIndices[k][0] + 2]) / k;
+                    deltaUV.x = (uvs[2 * tempIndices[k][k]] - uvs[2 * tempIndices[k][0]]) / k;
+                    deltaUV.y = (uvs[2 * tempIndices[k][k] + 1] - uvs[2 * tempIndices[k][0] + 1]) / k;
+                    for (var j = 1; j < k; j++) {
+                        tempIndices[k][j] = positions.length / 3;
+                        positions[positionPtr] = positions[3 * tempIndices[k][0]] + j * deltaPosition.x;
+                        normals[positionPtr++] = normals[3 * tempIndices[k][0]] + j * deltaNormal.x;
+                        positions[positionPtr] = positions[3 * tempIndices[k][0] + 1] + j * deltaPosition.y;
+                        normals[positionPtr++] = normals[3 * tempIndices[k][0] + 1] + j * deltaNormal.y;
+                        positions[positionPtr] = positions[3 * tempIndices[k][0] + 2] + j * deltaPosition.z;
+                        normals[positionPtr++] = normals[3 * tempIndices[k][0] + 2] + j * deltaNormal.z;
+                        uvs[uvPtr++] = uvs[2 * tempIndices[k][0]] + j * deltaUV.x;
+                        uvs[uvPtr++] = uvs[2 * tempIndices[k][0] + 1] + j * deltaUV.y;
+                    }
+                }
+                tempIndices[segments] = side[currentIndices[i + 1]][currentIndices[i + 2]];
+
+                // reform indices
+                indices.push(tempIndices[0][0], tempIndices[1][0], tempIndices[1][1]);
+                for (var k = 1; k < segments; k++) {
+                    for (var j = 0; j < k; j++) {
+                        indices.push(tempIndices[k][j], tempIndices[k + 1][j], tempIndices[k + 1][j + 1]);
+                        indices.push(tempIndices[k][j], tempIndices[k + 1][j + 1], tempIndices[k][j + 1]);
+                    }
+                    indices.push(tempIndices[k][j], tempIndices[k + 1][j], tempIndices[k + 1][j + 1]);
+                }
+            }
+
+            vertex_data.indices = indices;
+            vertex_data.applyToMesh(this);
+        }
+    }
+
+    /**
+     * Force adjacent facets to share vertices and remove any facets that have all vertices in a line
+     * This will undo any application of covertToFlatShadedMesh
+     * Introduced for use with Ammo.JSPlugin soft bodies
+     * Warning : the mesh is really modified even if not set originally as updatable. A new VertexBuffer is created under the hood each call.
+     */
+    public forceSharedVertices(): void {

```

Comment:
Is that different from convertToFlatShadedMesh ?

Reply:
Yes. It is the opposite. It would take a flatShadedMesh and revert it to its previous state. Any flat shaded mesh would be blown apart.

Reply:
Makes snes then we should not even mention Ammo here as this is useful for other scenarios
Also you have a type in covertToFlat

Reply:
Will remove Ammo comment.

Reply:
Will remove Ammo comment.

Reply:
Done